### PR TITLE
conditional switch to using oauth with fallback to api-token

### PR DESCRIPTION
**Pull request by sloev**

### CODE REVIEW

File: tap_pipedrive/tap.py [hunk 1]
```diff
@@ -244,17 +244,43 @@ def execute_stream_request(self, stream):
...
+        if "api_token" not in self.config:
+            payload = {
+                "grant_type": "refresh_token",
+                "refresh_token": self.config["refresh_token"],
+                "client_id": self.config["client_id"],
+                "client_secret": self.config["client_secret"],
+            }
+            resp = requests.post(OAUTH_URL + "/token", data=payload)
+            resp.raise_for_status()
+            if not resp:
```
Comment (paddie) [inline tap_pipedrive/tap.py]: is the `if not resp` not redundant with the `raise_for_status()` just above it? I believe it tests the same things, but I am not sure

File: tap_pipedrive/tap.py [hunk 2]
```diff
@@ -244,17 +244,43 @@ def execute_stream_request(self, stream):
...
-        }
-        if params:
-            _params.update(params)
-
         url = "{}/{}".format(BASE_URL, endpoint)
-        logger.debug("Firing request at {} with params: {}".format(url, _params))
+        logger.debug("Firing request at {} with params: {}".format(url, params))
 
-        return requests.get(url, headers=headers, params=_params)
+        # remove this case when all customers have authed with oauth
```
Comment (paddie) [inline tap_pipedrive/tap.py]: if you use `# TODO: ` then vscode has tooling to pick this up for you :)

File: tap_pipedrive/tap.py [hunk 3]
```diff
@@ -244,17 +244,43 @@ def execute_stream_request(self, stream):
...
+        if "api_token" in self.config:
+            params = {"api_token": self.config["api_token"], **params}
+            headers = {"User-Agent": self.config["user-agent"]}
+            response = requests.get(url, headers=headers, params=params)
```
Comment (paddie) [inline tap_pipedrive/tap.py:266]: I'd just add a return here and eliminate the nested else below:

```python
return requests.get(url, headers=headers, params=params)
```

**Review by paddie — APPROVED** [review]

File: tap_pipedrive/tap.py (hunk 3 above)
Comment (sloev) [reply tap_pipedrive/tap.py:266]: 👍

File: tap_pipedrive/tap.py (hunk 2 above)
Comment (sloev) [reply tap_pipedrive/tap.py]: 👍

File: tap_pipedrive/tap.py (hunk 1 above)
Comment (sloev) [reply tap_pipedrive/tap.py]: 👍 its true, no need for this

**Review by paddie — APPROVED** [review]

**Pull request merged**